### PR TITLE
Added release notes for v0.8

### PR DESCRIPTION
**Pull request by bocajnotnef**
Release notes! How fun!

### CODE REVIEW

**Comment by bocajnotnef** [comment]
@ctb @mr-c CR, please?

File: doc/RELEASE-0.8.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+============
+Release v0.8
+============
+
+We are pleased to announce the release of screed v0.8. screed is a database
```
Comment (mr-c) [inline doc/RELEASE-0.8.txt]: Screed should be capitalized, unlike 'khmer'.

File: doc/RELEASE-0.8.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+engine capable of storing and retrieving short-read sequence data. screed is
+designed to be fast and adaptable to different sequence file formats. This is
+the second release of screed that we consider to be stable and complete.
```
Comment (mr-c) [inline doc/RELEASE-0.8.txt]: There have been previous versions besides the two on PyPI

File: doc/RELEASE-0.8.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+screed project and a code of conduct for interacting with other contributors
+and project maintainers. 
+
+Features and Major Changes:
+ - bz2 compressed file streaming
+ - 'accuracy' attribute has been changed to 'quality'
+ - versioneer version string generator has been added
+ - dooxygen documentation engine has been added
+ - code of conduct and developer documentation has been added
+ - detailed changelog has been added
```
Comment (mr-c) [inline doc/RELEASE-0.8.txt]: List of contributors, links to issues closed?

Are there any known (outstanding) issues?

File: doc/RELEASE-0.8.txt (hunk 1 above)
Comment (bocajnotnef) [inline doc/RELEASE-0.8.txt]: Alright. It wasn't capitalized in the past release notes for screed.

On Sat, Mar 14, 2015, 00:31 Michael R. Crusoe <notifications@github.com>
wrote:

In doc/RELEASE-0.8.txt
<https://github.com/ged-lab/screed/pull/28#discussion_r26435272>:

> @@ -0,0 +1,20 @@ > +============ > +Release v0.8 > +============ > + >
+We are pleased to announce the release of screed v0.8. screed is a
database

Screed should be capitalized, unlike 'khmer'.


—
Reply to this email directly or view it on GitHub
<https://github.com/ged-lab/screed/pull/28/files#r26435272>.

Comment (ctb) [inline doc/RELEASE-0.8.txt]: On Sat, Mar 14, 2015 at 07:58:27AM -0700, bocajnotnef wrote:
> > @@ -0,0 +1,20 @@
> > +============
> > +Release v0.8
> > +============
> > +
> > +We are pleased to announce the release of screed v0.8. screed is a database
> 
> Alright. It wasn't capitalized in the past release notes for screed.

Indeed, it wasn't :)

**Comment by bocajnotnef** [comment]
@mr-c @ctb CR?

File: doc/RELEASE-0.8.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+We are pleased to announce the release of Screed v0.8. Screed is a database
+engine capable of storing and retrieving short-read sequence data. Screed is
+designed to be fast and adaptable to different sequence file formats. This 
+release of Screed we consider to be stable and complete.i
+
+This version of Screed contains developer documentation for contributing to the
+Screed project and a code of conduct for interacting with other contributors
+and project maintainers. 
+
```
Comment (mr-c) [inline doc/RELEASE-0.8.md]: Versioned link to docs

File: doc/RELEASE-0.8.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+## Features and Major Changes:
```
Comment (mr-c) [inline doc/RELEASE-0.8.md]: Give credit to authors per item in the list below

Comment (mr-c) [inline doc/RELEASE-0.8.md]: + links to PRs

File: doc/RELEASE-0.8.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+## Known Issues
+
+    * Screed records cannot be sliced. [Issue #768 (khmer repo)]
+    (https://github.com/ged-lab/khmer/issues/768) This will be included in the
+    next release.
+    * Screed self-tests do not use a temporary directory [Issue #748 (khmer
+    repo)] (https://github.com/ged-lab/khmer/issues/748) This is an issue if
+    the tests directory is read-only. 
+    * Screed does not support gzip file streaming. This is an issue with
+    python 2.x and will likely not be fixed in future releases.
```
Comment (mr-c) [inline doc/RELEASE-0.8.md]: Please capitalize the name of the Python programming language. It is a proper noun.

File: doc/RELEASE-0.8.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+## Features and Major Changes:
+ - bz2 compressed file streaming
+ - 'accuracy' attribute has been changed to 'quality'
+ - versioneer version string generator has been added
+ - dooxygen documentation engine has been added
+ - code of conduct and developer documentation has been added
+ - detailed changelog has been added
+
+## Closed Issues:
+    * [Screed gzip/bz2 file sniffing]
```
Comment (mr-c) [inline doc/RELEASE-0.8.md]: The extra indentation is causing this to render incorrectly. Also, why switch to '*' from '-' above?

**Comment by mr-c** [comment]
For future reference, here is how I reviewed this: I pulled up khmer's release notes and did a side-by-side compare using the 'rich diff' mode so I could check the formatting: https://github.com/ged-lab/screed/pull/28/files?short_path=9db2c2e#diff-9db2c2ef6ae81d5e481a172d607756c9

File: doc/RELEASE-0.8.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    * [screed returning the wrong version and breaking dev installs]
+      (https://github.com/ged-lab/khmer/issues/803)
+    * [make len(read) == len(read.sequence)]
+      (https://github.com/ged-lab/khmer/issues/705)
+    * [removed extraneous doc/scheme.txt]
+      (https://github.com/ged-lab/khmer/issues/800)
+
+## Known Issues
+
+    * Screed records cannot be sliced. [Issue #768 (khmer repo)]
```
Comment (mr-c) [inline doc/RELEASE-0.8.md]: Same formatting issue as above: excess indention turns this into a 'pre'formatted paragraph preserving all the markdown syntax

**Comment by mr-c** [comment]
Why the separate directory for old release notes?

File: doc/RELEASE-0.8.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+designed to be fast and adaptable to different sequence file formats. This 
+release of Screed we consider to be stable and complete.
+
+This version of Screed contains developer documentation for contributing to the
+Screed project and a code of conduct for interacting with other contributors
+and project maintainers. Documentation is available at
+http://screed.readthedocs.org/en/v0.8/ 
+
+## Features and Major Changes:
+ - bz2 compressed file streaming
```
Comment (mr-c) [inline doc/RELEASE-0.8.md]: And what about the credits for the new features, as I requested?

File: doc/RELEASE-0.8.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+ - 'accuracy' attribute has been changed to 'quality'
+ - Versioneer version string generator has been added
+ - Dooxygen documentation engine has been added
+ - code of conduct and developer documentation has been added
+ - detailed Changelog has been added
+
+## Closed Issues:
+ - [Screed gzip/bz2 file sniffing]
```
Comment (mr-c) [inline doc/RELEASE-0.8.md]: Please re-write the names of these to be more user friendly; many of the GitHub issues have opaque titles.

Note that for the khmer release notes I pull out the most important changes under 'New items of note', and I follow up with 'Notable' and 'Additional' bug fixed/issues closed.

File: doc/RELEASE-0.8.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+## Known Issues
+
+ - Screed records cannot be sliced. [Issue #768 (khmer repo)]
+    (https://github.com/ged-lab/khmer/issues/768) This will be included in the
+    next release.
+ - Screed self-tests do not use a temporary directory [Issue #748 (khmer
+    repo)] (https://github.com/ged-lab/khmer/issues/748) This is an issue if
+    the tests directory is read-only. 
+ - Screed does not support gzip file streaming. This is an issue with
+    Python 2.x and will likely not be fixed in future releases.
```
Comment (mr-c) [inline doc/RELEASE-0.8.md]: Link for this known issue?

File: doc/RELEASE-0.8.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+ - [normalize-by-median not accepting reads on stdin]
+      (https://github.com/ged-lab/khmer/issues/633) by @mr-c
+ - [screed returning the wrong version and breaking dev installs]
+      (https://github.com/ged-lab/khmer/issues/803) by @mr-c
+ - [make len(read) == len(read.sequence)]
+      (https://github.com/ged-lab/khmer/issues/705) by @gpratt
+ - [removed extraneous doc/scheme.txt]
+      (https://github.com/ged-lab/khmer/issues/800) by @bocajnotnef
+
+## Known Issues
```
Comment (mr-c) [inline doc/RELEASE-0.8.md]: Speaking of known issues; `doc/dev/known-issues.txt` should be updated to match this list and moved into the `doc/user` directory as we don't want to hide it.

**Comment by mr-c** [comment]
- [ ] put *all* release notes in `doc/release-notes`; don't rename this directory
- [ ] update `doc/dev/known-issues.txt` and move it to `doc/user/`

File: doc/release-notes/RELEASE-0.8.md [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+release of Screed we consider to be stable and complete.
+
+This version of Screed contains developer documentation for contributing to the
+Screed project and a code of conduct for interacting with other contributors
+and project maintainers. Documentation is available at
+http://screed.readthedocs.org/en/v0.8/ 
+
+## New items of note:
+
+This release successfully installs and passes its unit tests on Ubuntu 14.04 and the latest release of Mac OS X 10 "Yosemite". It also passes the khmer acceptance tests as per the [eelpond testing protocol.](https://github.com/ged-lab/literate-resting/blob/master/kp/README.txt)
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: No line may be longer than 80 characters

**Comment by bocajnotnef** [comment]
@mr-c @ctb CR?

File: doc/index.txt [hunk 14]
```diff
@@ -18,9 +18,10 @@ Contents:
 
    screed
    example
+   RELEASE-0.8
    
    dev/index
-   release-notes/index
+   old-release-notes/index
```
Comment (mr-c) [inline doc/index.txt]: I thought you got rid of 'old'?

**Comment by bocajnotnef** [comment]
@mr-c CR?

File: doc/user/known-issues.txt [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+Known Issues
+============
+
+This document details the known issues in the current release of screed. All
+issues for screed are tracked at https://github.com/ged-lab/khmer/issues
+
+List of known issues
+====================
+
+Screed does not support gzip file streaming. This is an issue with Python 2.x and will likely *not* be fixed in future releases. https://github.com/ged-lab/khmer/issues/700
```
Comment (mr-c) [inline doc/user/known-issues.txt]: 80 column wrap please.

You may enjoy adding this to your `~/.vimrc`

```
set textwidth=79  " lines longer than 79 columns will be broken
set colorcolumn=80
```

File: doc/user/known-issues.txt [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+
+Screed is overly tolerant of spaces in fast{q,a} which is against spec. https://github.com/ged-lab/khmer/issues/108
+
+Screed records cannot be sliced requiring un-Pythonic techniques to achieve the same behavior This will be included in the next release. https://github.com/ged-lab/khmer/issues/768
```
Comment (mr-c) [inline doc/user/known-issues.txt]: '...in a future release.'

File: doc/release-notes/RELEASE-0.8.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
+
+# Release v0.8
+
+We are pleased to announce the release of Screed v0.8. Screed is a database
+engine capable of storing and retrieving short-read sequence data. Screed is
+designed to be fast and adaptable to different sequence file formats. This 
+release of Screed we consider to be stable and complete.
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: I'd combine the 2nd and 3rd sentences and drop the last sentence.

File: doc/release-notes/RELEASE-0.8.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+acceptance tests as per the [eelpond testing protocol.]
+(https://github.com/ged-lab/literate-resting/blob/master/kp/README.txt)
+
+ - Screed now has automatic compression detection via magic bit sniffing gzip, 
+ bzip and zip files(from @mr-c)
+ - Screed now supports streaming of uncompressed FASTA and FASTQ files. Bzip2 
+ files can also be streamed but not gzip files. ([from @mr-c]
+ (https://github.com/ged-lab/screed/pull/11))
+ - Screed now has a Changelog, developer documentation and uses the 'quality' 
+ attribute over the 'accuracy' attribute in dealing with FASTQ reads. ([from
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: The accuracy-> quality change is quite a big deal as it changes the API. I would move it up to be the 2nd paragraph of the release notes. To be more precise I would say: "Screed has renamed the 'accuracy' attribute of read records to the name 'quality'; this API change will need to be adopted by all users who want to upgrade to this version. Unlike the khmer project, Screed does not yet adhere to semantic versioning. That will come with the 1.0 release"

File: doc/release-notes/RELEASE-0.8.md [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+
+ - Screed now has automatic compression detection via magic bit sniffing gzip, 
+ bzip and zip files(from @mr-c)
+ - Screed now supports streaming of uncompressed FASTA and FASTQ files. Bzip2 
+ files can also be streamed but not gzip files. ([from @mr-c]
+ (https://github.com/ged-lab/screed/pull/11))
+ - Screed now has a Changelog, developer documentation and uses the 'quality' 
+ attribute over the 'accuracy' attribute in dealing with FASTQ reads. ([from 
+ @bocajnotnef] (https://github.com/ged-lab/khmer/issues/625))
+ - Versioneer version string generator has been added ([from @bocajnotnef]
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: -> 'Versions are now autogenerated using git tags via versioneer.'

File: doc/release-notes/RELEASE-0.8.md [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+ (https://github.com/ged-lab/screed/pull/14))
+ - Dooxygen documentation engine has been added ([from @mr-c]
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: -> 'Documentation is now autogenerated using Doxygen.'

File: doc/release-notes/RELEASE-0.8.md [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+ (https://github.com/ged-lab/screed/pull/13))
+ - code of conduct and developer documentation has been added ([from @ctb, 
+ @mr-c and @bocajnotnef] (https://github.com/ged-lab/screed/pull/14))
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: already mentioned above; move the additional detail to the paragraph above.

**Comment by mr-c** [comment]
Looking good. You use three styles of linking to issues/pull requests; you should use just one and stick to it :-)

The style used in the khmer release notes mimics the autolinking GitHub applies to strings in the format "#n" where n is the number of the issue or pull request and the link is applied to the entire "#n" string.

**Comment by bocajnotnef** [comment]
@mr-c Had to do some funky things with the #n when the only relevant thing is the screed pull but other than that it should be good.

Blah. I like code far more than documentation.

**Comment by camillescott** [comment]
What sort of programmer prefers writing code over documenting it? What a weirdo ;)

**Comment by bocajnotnef** [comment]
@mr-c At somepoint here I'll manage to get everything. Good now?

File: doc/release-notes/RELEASE-0.8.md [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+ - Documentation is now autogenerated using Doxygen (from @mr-c in Screed/#13)
+
+## Notable bugs fixed/issues closed:
+ - [A khmer script was not accepting reads on the stdin]
+      (https://github.com/ged-lab/khmer/issues/633) by @mr-c
+ - [screed returning the wrong version and breaking dev installs]
+      (https://github.com/ged-lab/khmer/issues/803) by @mr-c
+
+
+## Known Issues
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: 3 known issues here, 4 in doc/user/known-issues.txt

File: doc/release-notes/RELEASE-0.8.md [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+protocol.]
+(https://github.com/ged-lab/literate-resting/blob/master/kp/README.txt)
+
+This release of screed has renamed the 'accuracy' attribute of read records to
+'quality;' this API change will need to be adopted by all users wanting to
+upgrade to this version. Unlike the khmer project, Screed is not currently
+under semantic versioning. It will be with the 1.0 release.
+
+ - Screed now has automatic compression detection via magic bit sniffing for 
+ gzip, bzip files (from @mr-c in #432)
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: "gzip and bzip compressed sequence files"

File: doc/release-notes/RELEASE-0.8.md [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+ - Screed now supports streaming of uncompressed FASTA and FASTQ files. Bzip2
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: "Screed now supports streaming reading of uncompressed FASTA and FASTQ formatted nucleotide sequence data."

File: doc/release-notes/RELEASE-0.8.md [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+ files can also be streamed but not gzip files. (from @mr-c, see #633)
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: "BZIP2 compressed FASTA and FASTQ formatted nucleotide sequence data can also be streamed but not GZIP compressed FASTA and FASTQ formatted nucleotide sequence data."

File: doc/release-notes/RELEASE-0.8.md [hunk 26]
```diff
@@ -0,0 +1,59 @@
...
+ - Screed now has a Changelog, developer documentation and a code of conduct
+ (from @ctb, @mr-c, @bocajnotnef in #625)
+ - Versions are now autogenerated using git tags via Versioneer (from
+ @bocajnotnef in Screed/#14)
+ - Documentation is now autogenerated using Doxygen (from @mr-c in Screed/#13)
+
+## Notable bugs fixed/issues closed:
+ - [A khmer script was not accepting reads on the stdin]
+      (https://github.com/ged-lab/khmer/issues/633) by @mr-c
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: Why are these links but not the others? Pick one style and stick with it :-)

File: doc/user/known-issues.txt [hunk 27]
```diff
@@ -0,0 +1,26 @@
+.. vim: set filetype=rst
+
+============
+Known Issues
+============
+
+This document details the known issues in the current release of screed. All
+issues for screed are tracked at https://github.com/ged-lab/khmer/issues
```
Comment (mr-c) [inline doc/user/known-issues.txt]: I would link directly to the screed issues at https://github.com/ged-lab/khmer/issues?q=label%3Ascreed

File: doc/release-notes/RELEASE-0.8.md [hunk 28]
```diff
@@ -0,0 +1,62 @@
...
+ gzip and bzip compressed files (from @mr-c in #432)
+ - Screed now supports streaming of uncompressed FASTA and FASTQ formatted 
+ nucleotide sequence data. Bzip2 compressed FASTA and FASTQ formatted
+ nucleotide sequence data can also be streamed but not GZIP compressed FASTA
+ and FASTQ formatted nucleotide sequence data. 
+ (from @mr-c, see #633)
+ - Screed now has a Changelog, developer documentation and a code of conduct
+ (from @ctb, @mr-c, @bocajnotnef in #625)
+ - Versions are now autogenerated using git tags via Versioneer (from
+ @bocajnotnef in Screed/#14)
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: why is this one `Screed/#14` versus all the others?

File: doc/release-notes/RELEASE-0.8.md [hunk 29]
```diff
@@ -0,0 +1,62 @@
...
+'quality;' this API change will need to be adopted by all users wanting to
+upgrade to this version. Unlike the khmer project, Screed is not currently
+under semantic versioning. It will be with the 1.0 release.
+
+ - Screed now has automatic compression detection via magic bit sniffing for 
+ gzip and bzip compressed files (from @mr-c in #432)
+ - Screed now supports streaming of uncompressed FASTA and FASTQ formatted 
+ nucleotide sequence data. Bzip2 compressed FASTA and FASTQ formatted
+ nucleotide sequence data can also be streamed but not GZIP compressed FASTA
+ and FASTQ formatted nucleotide sequence data.
```
Comment (camillescott) [inline doc/release-notes/RELEASE-0.8.md]: Here you use BZIP, but earlier you use bzip -- keep consistent

File: doc/release-notes/RELEASE-0.8.md [hunk 30]
```diff
@@ -0,0 +1,62 @@
...
+
+ - Screed now has automatic compression detection via magic bit sniffing for 
+ gzip and bzip compressed files (from @mr-c in #432)
+ - Screed now supports streaming of uncompressed FASTA and FASTQ formatted 
+ nucleotide sequence data. Bzip2 compressed FASTA and FASTQ formatted
+ nucleotide sequence data can also be streamed but not GZIP compressed FASTA
+ and FASTQ formatted nucleotide sequence data. 
+ (from @mr-c, see #633)
+ - Screed now has a Changelog, developer documentation and a code of conduct
+ (from @ctb, @mr-c, @bocajnotnef in #625)
```
Comment (camillescott) [inline doc/release-notes/RELEASE-0.8.md]: changelog

File: doc/release-notes/RELEASE-0.8.md [hunk 31]
```diff
@@ -0,0 +1,62 @@
...
+ - Versions are now autogenerated using git tags via Versioneer (from
```
Comment (camillescott) [inline doc/release-notes/RELEASE-0.8.md]: auto-generated

**Comment by camillescott** [comment]
Added a couple consistency things

File: doc/release-notes/RELEASE-0.8.md [hunk 32]
```diff
@@ -0,0 +1,62 @@
...
+ tests run from package-based installs to fail. This is being tracked in #748
+ - Screed does not support gzip file streaming. This is an issue with Python
+ 2.x and will likely *not* be fixed in future releases. This is being tracked
+ in #700
+ - Screed is overly tolerant of spaces in fast{a,q} which is against spec.
+ This is being tracked in #108
+ 
+## Contributors
+
+@ctb @mr-c @bocajnotnef @luizirber @brtaylor92 @grpratt
```
Comment (mr-c) [inline doc/release-notes/RELEASE-0.8.md]: I usually put this in the order that they are on https://github.com/ged-lab/screed/graphs/contributors?from=2014-05-19&to=2015-04-03&type=c

**Comment by bocajnotnef** [comment]
@mr-c Formal CR, please?

**Comment by mr-c** [comment]
Okay! Make a new RC and rerun the release checklist!

**Pull request merged**